tests/test_directory_sync.py

```diff
@@ -543,33 +482,10 @@ async def test_directory_user_groups_auto_pagination(
         groups = await self.directory_sync.list_groups()
         all_groups = []
 
-        async for group in groups.auto_paging_iter():
+        async for group in groups:
             all_groups.append(group)
 
         assert len(list(all_groups)) == len(mock_directory_groups_multiple_data_pages)
         assert (
             list_data_to_dicts(all_groups)
         ) == mock_directory_groups_multiple_data_pages
-
-    async def test_auto_pagination_honors_limit(
-        self,
-        mock_directories_multiple_data_pages,
-        mock_pagination_request_for_http_client,
-    ):
-        # TODO: This does not actually test anything about the limit.
```

Comment:
We lost this TODO. I still don't think we're testing inputs. Though, how *should* limit work with iteration? Is it a limit on the total number of returned records?

Reply:
This is a great question, I'm not sure. Worth us testing out, but also maybe playing with it in an example app/looking for prior art.